packages/aws-cdk-lib/aws-route53/lib/geo-location.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Routing based on geographical location.
+ */
+export class GeoLocation {
+  /**
+   * Geolocation resource record based on continent code.
+   * @param continentCode Continent.
+   * @returns Continent-based geolocation record
+   */
+  public static continent(continentCode: Continent) {
+    return new GeoLocation(continentCode, undefined, undefined);
+  }
+
+  /**
+   * Geolocation resource record based on country code.
+   * @param countryCode Two-letter, uppercase country code for the country.
+   * See ISO 3166-1-alpha-2 code on the *International Organization for Standardization* website
+   * @see https://docs.aws.amazon.com/Route53/latest/APIReference/API_GeoLocation.html#Route53-Type-GeoLocation-CountryCode
+   * @see https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+   * @returns Country-based geolocation record
+   */
+  public static country(countryCode: string) {
+    return new GeoLocation(undefined, countryCode, undefined);
+  }
+
+  /**
+   * Geolocation resource record based on subdivision code (e.g. state of the United States).
+   * @param subdivisionCode Code of the subdivision (e.g. state of the United States)
+   * @param countryCode Country code (ISO 3166-1-alpha-2) of this record, by default US (United States).
+   * @see https://pe.usps.com/text/pub28/28apb.htm
+   * @see https://docs.aws.amazon.com/Route53/latest/APIReference/API_GeoLocation.html#Route53-Type-GeoLocation-SubdivisionCode
+   */
+  public static subdivision(subdivisionCode: string, countryCode: string = 'US') {
+    return new GeoLocation(undefined, countryCode, subdivisionCode);
+  }
+
+  /**
+   * Default (wildcard) routing record if no specific geolocation record is found.
+   * @returns Wildcard routing record
+   */
+  public static default() {
+    return new GeoLocation(undefined, '*', undefined);
+  }
+
+  private static COUNTRY_REGEX = /(^[A-Z]{2}|^\*{1})$/;
+  private static COUNTRY_FOR_SUBDIVISION_REGEX = /\b(?:UA|US)\b/;
+  private static SUBDEVISION_REGEX = /^[A-Z0-9]{1,3}$/;
```

Comment:
Nit: Can we fix the spelling here to `SUBDIVISION_REGEX` instead of `SUBDEVISION_REGEX`